src/main/java/com/adorsys/ssi/sdjwt/SdJws.java

```diff
@@ -1,122 +1,123 @@
     protected SdJws(String jwsString) {
-        this.jwsInput = parse(jwsString);
-        this.payload = readPayload(jwsInput);
+        try {
+            this.jwsString = jwsString;
+            this.signedJwt = parse(jwsString);
+            this.payload = readPayload(signedJwt);
+        } catch (ParseException | IOException e) {
+            throw new RuntimeException(e);
+        }
     }
 
     // Constructor for signed JWS
-    protected SdJws(JsonNode payload, JWSInput jwsInput) {
+    protected SdJws(JsonNode payload, JWSObject signedJwt) {
         this.payload = payload;
-        this.jwsInput = jwsInput;
+        this.signedJwt = signedJwt;
+        this.jwsString = signedJwt.serialize();
     }
 
-    protected SdJws(JsonNode payload, SignatureSignerContext signer, String jwsType) {
+    protected SdJws(JsonNode payload, JWSSigner signer, String keyId, JWSAlgorithm jwsAlgorithm, String jwsType) {
         this.payload = payload;
-        this.jwsInput = sign(payload, signer, jwsType);
+        JWSHeader header = new JWSHeader.Builder(jwsAlgorithm).type(new JOSEObjectType(jwsType)).keyID(keyId).build();
+        this.signedJwt = new JWSObject(header, new Payload(Base64URL.encode(payload.toString())));
+        try {
+            this.signedJwt.sign(signer);
+        } catch (JOSEException e) {
+            throw new RuntimeException(e);
+        }
+        this.jwsString= signedJwt.serialize();
     }
-
-    protected static JWSInput sign(JsonNode payload, SignatureSignerContext signer, String jwsType) {
-        String jwsString = new JWSBuilder().type(jwsType).jsonContent(payload).sign(signer);
-        return parse(jwsString);
+    protected SdJws(Base64URL payloadBase64Url, JWSSigner signer, String keyId, JWSAlgorithm jwsAlgorithm, String jwsType) {
+        try {
+            this.payload = SdJwtUtils.mapper.readTree(payloadBase64Url.decode());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        JWSHeader header = new JWSHeader.Builder(jwsAlgorithm).type(new JOSEObjectType(jwsType)).keyID(keyId).build();
+        this.signedJwt = new JWSObject(header, new Payload(payloadBase64Url));
+        try {
+            this.signedJwt.sign(signer);
+        } catch (JOSEException e) {
+            throw new RuntimeException(e);
+        }
+        this.jwsString= signedJwt.serialize();
     }
 
-    public void verifySignature(SignatureVerifierContext verifier) throws VerificationException {
-        Objects.requireNonNull(verifier, "verifier must not be null");
-        try {
-            if (!verifier.verify(jwsInput.getEncodedSignatureInput().getBytes(StandardCharsets.UTF_8), jwsInput.getSignature())) {
-                throw new VerificationException("Invalid jws signature");
-            }
-        } catch (Exception e) {
-            throw new VerificationException(e);
+    public void verifySignature(JWSVerifier verifier) throws JOSEException, JOSEException {
```

Comment:
method does not throws JOSEException

Reply:
Fixed with [#3a6a88a](https://github.com/adorsys/sd-jwt/pull/3/commits/3a6a88a760dd18bff56bbcb487bcfada956378b1)